src/DotNetty.Common/Utilities/StringUtil.cs

```diff
@@ -0,0 +1,331 @@
+            }
+        }
+
+        /// <summary>
+        /// Splits the specified {@link String} with the specified delimiter in maxParts maximum parts.
+        /// This operation is a simplified and optimized
+        /// version of {@link String#split(String, int)}.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="delim"></param>
+        /// <param name="maxParts"></param>
+        /// <returns></returns>
+        public static string[] Split(string value, char delim, int maxParts)
+        {
+            int end = value.Length;
+            var res = new List<string>();
+
+            int start = 0;
+            int cpt = 1;
+            for (int i = 0; i < end && cpt < maxParts; i++)
+            {
+                if (value[i] == delim)
+                {
+                    if (start == i)
+                    {
+                        res.Add(string.Empty);
+                    }
+                    else
+                    {
+                        res.Add(value.Substring(start, i));
+                    }
+                    start = i + 1;
+                    cpt++;
+                }
+            }
+
+            if (start == 0)
+            {
+                // If no delimiter was found in the value
+                res.Add(value);
+            }
+            else
+            {
+                if (start != end)
+                {
+                    // Add the last element if it's not empty.
+                    res.Add(value.Substring(start, end));
+                }
+                else
+                {
+                    // Truncate trailing empty elements.
+                    for (int i = res.Count - 1; i >= 0; i--)
+                    {
+                        if (res[i] == "")
+                        {
+                            res.Remove(res[i]);
+                        }
+                        else
+                        {
+                            break;
+                        }
+                    }
+                }
+            }
+
+            return res.ToArray();
+        }
+
+        /// <summary>
+        /// Get the item after one char delim if the delim is found (else null).
+        /// This operation is a simplified and optimized
+        /// version of {@link String#split(String, int)}.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="delim"></param>
+        /// <returns></returns>
+        public static string SubstringAfter(this string value, char delim)
+        {
+            int pos = value.IndexOf(delim);
+            return pos >= 0 ? value.Substring(pos + 1) : null;
+        }
+
+        /// <summary>
+        ///  Converts the specified byte value into a 2-digit hexadecimal integer.
+        /// </summary>
+        public static string ByteToHexStringPadded(int value)
+        {
+            return Byte2HexPad[value & 0xff];
+        }
+
+        //    /**
```

Comment:
I don't know if we have anything like Appendable in .Net, but i will mark it for future porting.